rtengine/dcraw.cc

```diff
@@ -7647,37 +7647,57 @@ void CLASS parse_fuji (int offset)
 
   fseek (ifp, offset, SEEK_SET);
   entries = get4();
+  int read_crop_c = 0;                                                  // RT
+  bool read_crop_dimensions = false;                                    // RT
   if (entries > 255) return;
   while (entries--) {
     tag = get2();
     len = get2();
     save = ftell(ifp);
+    // tag 0x100 = 256 RawImageFullSize                                    RT
     if (tag == 0x100) {
       raw_height = get2();
       raw_width  = get2();
     } else if (tag == 0x121) {
       height = get2();
       if ((width = get2()) == 4284) width += 3;
+      // tag 0x130 = 304 FujiLayout                                        RT
     } else if (tag == 0x130) {
       fuji_layout = fgetc(ifp) >> 7;
       fuji_width = !(fgetc(ifp) & 8);
+      // tag 0x131 = 305 XTransLayout                                      RT
     } else if (tag == 0x131) {
       filters = 9;
       FORC(36) xtrans_abs[0][35-c] = fgetc(ifp) & 3;
     } else if (tag == 0x2ff0) {
       FORC4 cam_mul[c ^ 1] = get2();
-    } else if (tag == 0xc000 && len > 20000) {
+    } else if (tag == 0xc000 && len > 20000 && !read_crop_dimensions) { // RT
```

Comment:
Since you're reading the crop dimensions into its own struct, I think it would be better to to update the width and height after the loop with the crop's values if a crop mode is detected (tag 0x104d, [reference](https://exiftool.org/TagNames/FujiFilm.html)). That will also be a good place to set the `left_margin` and `right_margin`.